easybuild/easyconfigs/a/ADIOS/ADIOS-1.13.1-foss-2020a-Python-3.8.2.eb

```diff
@@ -0,0 +1,68 @@
+# Based on ADIOS-1.13.1-foss-2019a-Python-2.7.15.eb
+# Updated to foss-2020a toolchain and switched to ConfigureMake
+# Mini-XML updated to latest 2.x. version
+# Author: J. Sassmannshausen (Imperial College London/UK)
+
+easyblock = 'ConfigureMake'
+
+name = 'ADIOS'
+version = '1.13.1'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'https://www.olcf.ornl.gov/center-projects/adios/'
+description = """The Adaptable IO System (ADIOS) provides a simple,
+flexible way for scientists to describe the data in their code that may
+need to be written, read, or processed outside of the running
+simulation."""
+
+toolchain = {'name': 'foss', 'version': '2020a'}
+toolchainopts = {'usempi': True, 'pic': True}
+
+github_account = 'ornladios'
+source_urls = [GITHUB_SOURCE]
+sources = ['v%(version)s.tar.gz']
+patches = [
+    '%(name)s-%(version)s_force_use_of_mpi.patch',
+    '%(name)s-%(version)s_fix_missing_thread_lib.patch',
+    '%(name)s-%(version)s_fix_search_for_szip.patch',
+]
+checksums = [
+    'b1c6949918f5e69f701cabfe5987c0b286793f1057d4690f04747852544e157b',  # v1.13.1.tar.gz
+    '81b9b0a77b23d6137f08900a3ecda4471b806c384cf4ad19c4370fc7ca8d9a82',  # ADIOS-1.13.1_force_use_of_mpi.patch
+    'a66fab38e5daf40978f9e961d810f9cbf189de8db924a403ae42a01d405f6fdc',  # ADIOS-1.13.1_fix_missing_thread_lib.patch
+    '2e21a5041822c8b57554eb977a3135637c2714d377eee3b0194d377f1458cdab',  # ADIOS-1.13.1_fix_search_for_szip.patch
+]
+
+builddependencies = [
+    ('Autotools', '20180311'),
+]
+
+dependencies = [
+    ('Python', '3.8.2'),
+    ('SciPy-bundle', '2020.03', versionsuffix),
+    ('zlib', '1.2.11'),
+    ('bzip2', '1.0.8'),
+    ('Szip', '2.1.1'),
+    ('lz4', '1.9.2'),
+    ('netCDF', '4.7.4'),
+    ('Mini-XML', '2.12'),
+]
+
+preconfigopts = ' autoreconf -i && '
+configopts = '--with-zlib=$EBROOTZLIB -with-bzip2=$EBROOTBZIP2 --with-netcdf==$EBROOTNETCDF '
+configopts += '--with-lz4=$EBROOTLZ4 --with-mxml=$EBROOTMINIMINXML --with-phdf5=$EBROOTHDF5 '
+
+fix_python_shebang_for = ['bin/gpp.py']
+
+runtest = ' check'
+
+sanity_check_paths = {
+    'files': ['bin/adios_list_methods', 'bin/bpappend'],
+    'dirs': ['etc/skel/templates', 'lib/python'],
+}
+
```

Comment:
Is it possible to add some sanity check commands for both ADIOS versions?

Reply:
Good enough?

```
$ adios_list_methods
Available write methods (in XML <method> element or in adios_select_method()):
    "MPI"
    "POSIX"
    "PHDF5"
    "MPI_LUSTRE"
    "MPI_AGGREGATE"
    "VAR_MERGE"
Available read methods (constants after #include "adios_read.h"):
    ADIOS_READ_METHOD_BP (=0)
    ADIOS_READ_METHOD_BP_AGGREGATE (=1)
Available data transformation methods (in XML transform tags in <var> elements):
    "none"      : No data transform
    "identity"  : Identity transform
    "zlib"      : zlib compression
    "bzip2"     : bzip2 compression
    "zfp"       : zfp compression
    "lz4"       : lz4 compression
Available query methods (in adios_query_set_method()):
    ADIOS_QUERY_METHOD_MINMAX (=0)

```

```
$ bpappend -h
Usage: bpappend [-h | --help] [-v | --verbose] splitfile appendfile

Append timesteps into a time-stepped BP file from another file BP file splitted by bpsplit.
splitfile will be appended to appendfile.

  --help | -h     Print this help.
  --verbose | -v  Log activity about what this program is doing.
                  Extra -v increases the number of messages.

$ echo $?
0
```

Reply:
For reasons I don't understand, I don't seem to be able to PR my fork into @sassy-crick's fork.
https://github.com/hattom/easybuild-easyconfigs/tree/20220816220336_new_pr_ADIOS1131

Reply:
PR now created with some URL magic.

Reply:
Yes, those checks would be fine! @sassy-crick assuming you also agree, could you merge https://github.com/sassy-crick/easybuild-easyconfigs/pull/9? Then the changes will show up here.

Reply:
I just have merged sassy-crick#9. Is there anything else i need to do?

Reply:
I should add the same test to the other file, will make a `#10` PR on your fork.

Reply:
Have made: https://github.com/sassy-crick/easybuild-easyconfigs/pull/10